dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -14,15 +14,20 @@ import com.datadog.android.rum.RumMonitor
 import com.datadog.android.rum.RumResourceKind
 import com.datadog.android.rum.internal.domain.RumEvent
 import com.datadog.android.rum.internal.domain.RumEventData
+import com.datadog.android.rum.internal.domain.RumEventSerializer
+import java.lang.ref.WeakReference
 import java.util.UUID
 
 internal class DatadogRumMonitor(
     private val timeProvider: TimeProvider,
     private val writer: Writer<RumEvent>
 ) : RumMonitor {
 
-    private val activeView = mutableMapOf<Any, RumEvent>()
-    private val activeResources = mutableMapOf<Any, RumEvent>()
+    private var activeViewEvent: RumEvent? = null
+    private var activeViewData: RumEventData.View? = null
```

Comment:
why do you have to keep this here ? Isn't this referenced already in the `activeViewEvent` ? I think we can simplify a bit this logic by removing this reference.

Reply:
we could but because the `activeViewEvent.data` can be any type, it'll mean doing casts and checks every time you want to manipulate it.